README.md

```diff
@@ -179,6 +187,23 @@ $.Function :: Type
 
 Type comprising every Function value.
 
+#### `NonZeroFiniteNumber`
+
+```haskell
+$.NonZeroFiniteNumber :: Type
+```
+
+Type comprising every Number value except `Infinity`, `-Infinity`, `NaN`, `0` 
+and `-0`. 
```

Comment:
Yeah, I wasn't sure how to handle forward references in the docs for types that depended on `ValidNumber` (because that appears last in the list) so I just defined everything in line. This is a better solution.

:zap: